File: 直接插入排序/InsertSort.java
public class InsertSort {

    public static void main(String[] args) {
        int l[] = {0, 5, 3, 4, 6, 2};
        int i, j;
        for (i = 2; i < l.length; i++) {
            if (l[i] < l[i - 1]) {
                l[0] = l[i];
                for (j = i - 1; l[j] > l[0]; j--) {  //找出比标志位更大的数往前移
                    l[j + 1] = l[j];
                }
                l[j + 1] = l[0];
            }
        }
    }
}

File: 直接插入排序/InsertSort.py
class insertsort(object):

    l=[0,5,3,4,6,2]

    def insert_sort(l):
        i = 2
        j = 0
        while i < len(l):
            if l[i] < l[i-1]:
                l[0] = l[i]
                j = i -1
                while l[j] > l[0]:
                    l[j+1] = l[j]
                    j = j -1
                l[j+1] = l[0]
            i = i + 1
        print(l)

    if __name__ == '__main__':
        insert_sort(l)
        input()
